include/Song.h

```diff
@@ -228,6 +228,16 @@ class EXPORT Song : public TrackContainer
 		return m_modified;
 	}
 
+	bool isModifiedSinceAutoSave()
+	{
+		return m_modifiedSinceAutoSave;
+	}
+
+	void setUnmodifiedSinceAutoSave()
+	{
+		m_modifiedSinceAutoSave = false;
```

Comment:
Food for thought... It seems if we switched `m_modified` to a timestamp, we could still live with only one variable.